File: fb-customers/src/main/java/de/effectivetrainings/billy/customermanagement/repository/domain/Customer.java
package de.effectivetrainings.billy.customermanagement.repository.domain;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;

/**
 *
 */
@Getter
@NoArgsConstructor
@EqualsAndHashCode
public class Customer {

    public static final String AGGREGATE_NAME = "customer.customers";

    //TODO events - implement mongo sequence

    private String customerId;
    private String customerNumber;
    private String organization;

    public Customer(String customerId, String customerNumber, String organization) {
        this.customerId = customerId;
        this.customerNumber = customerNumber;
        this.organization = organization;
    }


}

File: services/fb-web/app/src/customers/Customer.js
module.exports = function(customerId, customerNumber, organization, firstName, lastName) {
    this.customerId = customerId;
    this.firstName = firstName;
    this.lastName = lastName;
    this.customerNumber = customerNumber;
    this.organization = organization;

    this.toJSON = function() {
        return  {
            'FIRST_NAME' : this.firstName,
            'LAST_NAME' : this.lastName,
            'CUSTOMER_ID' : this.customerId,
            'CUSTOMER_NUMBER' : this.customerNumber,
            'ORGANIZATION' : this.organization
        };
    }

};